packages/react-core/src/next/components/Select/SelectOption.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { css } from '@patternfly/react-styles';
+import { MenuItemProps, MenuItem } from '../../../components/Menu';
+
+export interface SelectOptionProps extends Omit<MenuItemProps, 'ref'> {
+  /** Anything which can be rendered in a select option */
+  children?: React.ReactNode;
+  /** Classes applied to root element of select option */
+  className?: string;
+  /** Identifies the component in the Menu onSelect or onActionClick callback */
+  itemId?: any;
+  /** Flag indicating the option has a checkbox */
+  hasCheck?: boolean;
+  /** Render option as disabled */
```

Comment:
Nit: just to follow the other boolean props, we could do "Flag indicating the option is disabled" instead.